src/main/java/org/wildfly/security/auth/spi/AuthenticatedRealmIdentity.java

```diff
@@ -35,6 +36,13 @@
     Principal getPrincipal();
 
     /**
+     * Returns the roles for this identity.
+     *
+     * @return A string set containing the roles for this identity or an empty set if this identity has no roles.
```

Comment:
Wish I had that when I joined JBoss. Really, really nice. Now I have a proven checklist to follow when reviewing my code.

Thanks for the heads up @dmlloyd !